lib/mail/parsers/rfc_2822.ex

```diff
@@ -307,8 +310,8 @@ defmodule Mail.Parsers.RFC2822 do
   defp parse_recipient_value(value) do
     Regex.scan(~r/\s*"?(.*?)"?\s*?<?([^\s]+@[^\s>]+)>?,?/, String.replace(value, "\"", ""))
```

Comment:
Apparently comma is allowed in email address, under specific conditions. 
https://stackoverflow.com/a/38787343

Probably just this regex capture is wrong but I refuse to understand it :smile: 

Let's just maybe instead of `String.replace(address, ",", "")`, do comma cleanup only if it's last character

Reply:
@dziemian007 my regex adjustment will not affect commas inside of the address, only the ones at the end

Reply:
Yes only at the end

Reply:
ok, so I guess this one will be better if we only want to check for comma at the end:
```
~r/\s*"?(.*?)"?\s*?<?([^\s]+@[^\s>]+[^,> ])>?,?/
```

Reply:
but TBH I think either way it's fine :D